webui/react/src/components/kit/Select.tsx

```diff
@@ -0,0 +1,159 @@
+  let count = 0;
+
+  if (options) return options.length;
+  if (!children) return count;
+
+  if (Array.isArray(children)) {
+    count += children
+      .map((child) => countOptions(child, options))
+      .reduce((acc, count) => acc + count, 0);
+  }
+
+  const childType = (children as React.ReactElement).type;
+  const childProps = (children as React.ReactElement).props;
+  const childList = (childProps as React.ReactPortal)?.children;
+  if (childType === Option) count++;
+  if (childType === OptGroup && childList) count += countOptions(childList, options);
+
+  return count;
+};
+
+const Select: React.FC<React.PropsWithChildren<Props>> = forwardRef(function Select(
+  {
+    allowClear,
+    defaultValue,
+    disabled,
+    disableTags = false,
+    enableSearchFilter = true,
+    filterOption,
+    filterSort,
+    id,
+    label,
+    mode,
+    onChange,
+    onBlur,
+    onDeselect,
+    onSearch,
+    onSelect,
+    options,
+    placeholder,
+    placement,
+    showSearch,
+    value,
+    maxTagPlaceholder,
+    children,
+  }: React.PropsWithChildren<Props>,
+  ref?: React.Ref<RefSelectProps>,
+) {
+  const [isOpen, setIsOpen] = useState(false);
+  const classes = [css.base];
+
+  if (disableTags) classes.push(css.disableTags);
+  if (mode === 'multiple') {
+    classes.push(css.multiple);
+  }
+  const optionsCount = useMemo(() => countOptions(children, options), [children, options]);
+
+  const [maxTagCount, maxTagPlaceholderValue] = useMemo((): [
+    number | undefined | 'responsive',
+    string,
+  ] => {
+    if (!disableTags) return [undefined, maxTagPlaceholder ? maxTagPlaceholder : ''];
+    const count = Array.isArray(value) ? value.length : value ? 1 : 0;
+    const itemLabel = 'selected';
+    const placeholder = count === optionsCount ? 'All' : `${count} ${itemLabel}`;
+    return isOpen ? [0, ''] : [0, placeholder];
+  }, [disableTags, isOpen, optionsCount, maxTagPlaceholder, value]);
+
+  const handleDropdownVisibleChange = useCallback((open: boolean) => {
+    setIsOpen(open);
+  }, []);
+  const handleFilter = useCallback((search: string, option?: DefaultOptionType): boolean => {
+    let label: string | null = null;
+    if (option?.children) {
+      if (Array.isArray(option.children)) {
+        label = option.children.join(' ');
+      } else if (option.label) {
+        label = option.label.toString();
+      } else if (typeof option.children === 'string') {
+        label = option.children;
+      }
+    }
+
+    return !!label && label.toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
+  }, []);
+  return (
+    <div className={classes.join(' ')}>
+      {label && <Label type={LabelTypes.TextOnly}>{label}</Label>}
+      <AntdSelect
+        allowClear={allowClear}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        dropdownMatchSelectWidth={250}
+        filterOption={
+          enableSearchFilter || filterOption ? (filterOption ? filterOption : handleFilter) : true
```

Comment:
fix: I'm not sure what this is doing, but I think it reduces to `enableSearchFilter || filterOption ?? true`? The nested ternary is throwing me off, it doesn't seem like it would ever return `handleFilter`

Reply:
@EmilyBonar 

The logic here is supposed to be:

```
if (enableSearchFilter || filterOption) 
    if filterOption
        return filterOption
   else
        return handleFilter
else
   // do not filter any values
   return true  
```

basically we need this because you can `enableSearchFilter` (enable the ability to search) without defining a function to filter on `filterOption`, in that specific case we just return the default filtering function `handleFilter` 

I am open to other ways to handle `enableSearch` and `filterOption` but this did seem like the simplest way.

Reply:
Ah, I see. In that case, I think this does the same thing but is a little more clear: `filterOption ?? (enableSearchFilter ? handleFilter : true)`